### PR TITLE
IP space resource and data source implementation

### PR DESCRIPTION
**Pull request by mathewab**

### CODE REVIEW

File: internal/service/ipam/model_ipamsvc_inherited_dhcp_option_item.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+func (m *IpamsvcInheritedDHCPOptionItemModel) Flatten(ctx context.Context, from *ipam.IpamsvcInheritedDHCPOptionItem, diags *diag.Diagnostics) {
+	if from == nil {
+		return
+	}
+	if m == nil {
+		*m = IpamsvcInheritedDHCPOptionItemModel{}
+	}
+	m.Option = FlattenIpamsvcOptionItem(ctx, from.Option, diags)
+	m.OverridingGroup = flex.FlattenStringPointer(from.OverridingGroup)
+
```
Comment (venkat-iblox) [inline internal/service/ipam/model_ipamsvc_inherited_dhcp_option_item.go:81]: We should get rid of this space.

**Comment by venkat-iblox** [comment]
We should also run `make fmt` on the implemented code

**Comment by mathewab** [comment]
> We should also run `make fmt` on the implemented code

The generator runs goimports which includes gofmt. But yes, we should add a github action to check for fmt.

File: internal/service/ipam/model_ipamsvc_inherited_dhcp_option_item.go (hunk 1 above)
Comment (mathewab) [reply internal/service/ipam/model_ipamsvc_inherited_dhcp_option_item.go]: Fixed, here and in generator too

**Review by venkat-iblox — APPROVED** [review]

**Pull request merged**